x-pack/plugins/fleet/public/applications/fleet/sections/settings/components/outputs_table/index.tsx

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import styled from 'styled-components';
+import { EuiBasicTable, EuiButtonIcon, EuiFlexGroup, EuiFlexItem, EuiIcon } from '@elastic/eui';
+import type { EuiBasicTableColumn } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { useLink } from '../../../../hooks';
+import type { Output } from '../../../../types';
+
+import { DefaultBadges } from './badges';
+
+export interface OutputsTableProps {
+  outputs: Output[];
+  deleteOutput: (output: Output) => void;
+}
+
+const NameFlexItemWithMaxWidth = styled(EuiFlexItem)`
+  max-width: 250px;
+`;
+
+// Allow child to be truncated
+const FlexGroupWithMinWidth = styled(EuiFlexGroup)`
+  min-width: 0px;
+`;
+
+function displayOutputType(type: string) {
+  switch (type) {
+    case 'elasticsearch':
+      return i18n.translate('xpack.fleet.settings.outputsTable.elasticsearchTypeLabel', {
+        defaultMessage: 'Elasticsearch',
+      });
+    default:
+      return type;
+  }
+}
+
+export const OutputsTable: React.FunctionComponent<OutputsTableProps> = ({
+  outputs,
+  deleteOutput,
+}) => {
+  const { getHref } = useLink();
+
+  const columns = useMemo((): Array<EuiBasicTableColumn<Output>> => {
+    return [
+      {
+        render: (output: Output) => (
+          <EuiFlexGroup alignItems="center" gutterSize="xs">
+            <NameFlexItemWithMaxWidth grow={false}>
+              <p title={output.name} className={`eui-textTruncate`}>
+                {output.name}
+              </p>
+            </NameFlexItemWithMaxWidth>
+            {output.is_preconfigured && (
+              <EuiFlexItem grow={false}>
+                <EuiIcon type="lock" />
+              </EuiFlexItem>
+            )}
+          </EuiFlexGroup>
+        ),
+        width: '288px',
```

Comment:
could there be any `eui theme `values for these hardcoded values?

Reply:
I do not think we have eui values here that will work for this table, I am still going to double check